lib/new_relic/agent/instrumentation/aws_sqs/instrumentation.rb

```diff
@@ -0,0 +1,87 @@
+# This file is distributed under New Relic's license terms.
+# See https://github.com/newrelic/newrelic-ruby-agent/blob/main/LICENSE for complete details.
+# frozen_string_literal: true
+
+module NewRelic::Agent::Instrumentation
+  module AwsSqs
+    MESSAGING_LIBRARY = 'SQS'
+
+    def send_message_with_new_relic(*args)
+      segment = nil
+      begin
+        queue_name = get_queue_name(args[0])
+        segment = NewRelic::Agent::Tracer.start_message_broker_segment(
+          action: :produce,
+          library: MESSAGING_LIBRARY,
+          destination_type: :queue,
+          destination_name: queue_name
+        )
+        add_aws_attributes(segment, queue_name, args[0])
+      rescue => e
+        NewRelic::Agent.logger.error('Error starting message broker segment in Aws::SQS::Client#send_message ', e)
+      end
+      NewRelic::Agent::Tracer.capture_segment_error(segment) do
```

Comment:
Currently, fully 3 operations are wrapped in the initial `begin/rescue` block:

- `get_queue_name`
- `NewRelic::Agent::Tracer.start_message_broker_segment`
- `add_aws_attributes`

Two things:

1. Is there any need to individually wrap each call in its own `begin/rescue` block, possibly by using 3 dedicated methods?

2. This leads me to believe that it is entirely possible to reach this `capture_segment_error` line with a `segment` variable that is `nil`. `capture_segment_error` itself doesn't appear to check `segment` for `nil`, so I think we need to guard for `nil` before calling it.

Reply:
for #1, The purpose of the begin/rescue is if any issues happen as part of our instrumentation, the gem will still make the original call it intended to make, so anything wrong happening for us won't break the customers application. I'm not sure why we'd need to wrap each line in its own rescue, is there something specific you are thinking of that would make that necessary? 

For #2, we don't need to check for nil. If segment is nil, it basically is just a noop method, since the only time segment is used it first checks to make sure it's the correct class, so if it's nil, it will simply skip it instead.